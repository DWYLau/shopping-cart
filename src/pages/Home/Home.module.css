@import url("https://fonts.googleapis.com/css2?family=Montserrat:ital,wght@1,100..900&display=swap");

body {
  overflow-x: hidden; /* Hide scrollbars */
}

.banner {
  margin-top: 3rem;
  background: #ccff00;
  padding: 3rem 0;
  width: 100%;
  font-family: "Montserrat", sans-serif;
  font-optical-sizing: auto;
  font-weight: 600;
  font-style: italic;
  font-size: 3rem;
  text-align: center;
}

.image-container {
  display: flex;
  justify-content: center;
  align-items: center;
  margin-top: 3rem;
}

.image {
  height: 40vh;
}

.model-container {
  margin-top: 3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 1rem;
}

.card {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 1rem;
}

.card h2 {
  text-transform: uppercase;
  font-family: "Montserrat", sans-serif;
  font-optical-sizing: auto;
  font-weight: 600;
  font-size: 1.3rem;
}

.card img {
  height: 400px;
  width: 270px;
}
