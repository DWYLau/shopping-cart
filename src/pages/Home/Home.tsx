import styles from "./Home.module.css"
import Zoolander from "../../assets/images/zoolander.webp"
import model1 from "../../assets/images/model1.jpg"
import model2 from "../../assets/images/model2.jpg"
import model3 from "../../assets/images/model3.jpg"
import model4 from "../../assets/images/model4.jpg"
import model5 from "../../assets/images/model5.jpg"
import model6 from "../../assets/images/model6.jpg"
import { Link } from "react-router-dom"

function Home() {
  return (
    <section>
      <div className={styles.banner}>Up to 50% discount on selected items!</div>
      <div className={styles["image-container"]}>
        <img
          className={styles.image}
          src={Zoolander}
          alt='Model turning around'
        />
      </div>
      <div className={styles["model-container"]}>
        <div className={styles.card}>
          <img src={model1} alt='Model' />
          <h2>Latest Trends</h2>
        </div>
        <div className={styles.card}>
          <img src={model2} alt='Model' />
          <h2>Modern Looks</h2>
        </div>
        <div className={styles.card}>
          <img src={model3} alt='Model' />
          <h2>Stylish Vogue</h2>
        </div>
        <div className={styles.card}>
          <img src={model4} alt='Model' />
          <h2>Formal Fashion</h2>
        </div>
      </div>
      <div className={styles["second-banner"]}>Special Spring Sale!</div>
      <div className={styles["flex-container"]}>
        <div className={styles["model-container2"]}>
          <img src={model5} alt='Model' />
          <h2>Classical</h2>
          <Link to='products'>
            <button>Products</button>
          </Link>
        </div>
        <div className={styles["model-container2"]}>
          <img src={model6} alt='Model' />
          <h2>Contemporary</h2>
          <Link to='products'>
            <button>Products</button>
          </Link>
        </div>
      </div>
    </section>
  )
}

export default Home
