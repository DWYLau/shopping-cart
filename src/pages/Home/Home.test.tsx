import Home from "./Home"
import { BrowserRouter } from "react-router-dom"
import { describe, it, expect } from "vitest"
import { getByText, render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"

describe("Home page component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )

    expect(container).toMatchSnapshot()
  })

  it("renders banner text", () => {
    const { container } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )

    getByText(container, "Up to 50% discount on selected items!")
  })

  it("product button changes page", async () => {
    const user = userEvent.setup()

    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    )

    const buttons = screen.getAllByRole("button", {
      name: "Products",
    }) as HTMLButtonElement[]

    for (const button of buttons) {
      await user.click(button)
    }

    expect(window.location.href).toBe("http://localhost:3000/products")
  })
})
