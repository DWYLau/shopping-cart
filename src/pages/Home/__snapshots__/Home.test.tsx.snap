// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page component > renders correctly 1`] = `
<div>
  <section>
    <div
      class="_banner_da09e4"
    >
      Up to 50% discount on selected items!
    </div>
    <div
      class="_image-container_da09e4"
    >
      <img
        alt="Model turning around"
        class="_image_da09e4"
        src="/src/assets/images/zoolander.webp"
      />
    </div>
    <div
      class="_model-container_da09e4"
    >
      <div
        class="_card_da09e4"
      >
        <img
          alt="Model"
          src="/src/assets/images/model1.jpg"
        />
        <h2>
          Latest Trends
        </h2>
      </div>
      <div
        class="_card_da09e4"
      >
        <img
          alt="Model"
          src="/src/assets/images/model2.jpg"
        />
        <h2>
          Modern Looks
        </h2>
      </div>
      <div
        class="_card_da09e4"
      >
        <img
          alt="Model"
          src="/src/assets/images/model3.jpg"
        />
        <h2>
          Stylish Vogue
        </h2>
      </div>
      <div
        class="_card_da09e4"
      >
        <img
          alt="Model"
          src="/src/assets/images/model4.jpg"
        />
        <h2>
          Formal Fashion
        </h2>
      </div>
    </div>
    <div
      class="_second-banner_da09e4"
    >
      Special Spring Sale!
    </div>
    <div
      class="_flex-container_da09e4"
    >
      <div
        class="_model-container2_da09e4"
      >
        <img
          alt="Model"
          src="/src/assets/images/model5.jpg"
        />
        <h2>
          Classical
        </h2>
        <a
          href="/products"
        >
          <button
            class="_button_da09e4"
          >
            Products
          </button>
        </a>
      </div>
      <div
        class="_model-container2_da09e4"
      >
        <img
          alt="Model"
          src="/src/assets/images/model6.jpg"
        />
        <h2>
          Contemporary
        </h2>
        <a
          href="/products"
        >
          <button
            class="_button_da09e4"
          >
            Products
          </button>
        </a>
      </div>
    </div>
    <div
      class="_banner_da09e4"
    >
      Latest Fashion Trends! Get them here!
    </div>
    <div
      class="_flex-container2_da09e4"
    >
      <div
        class="_social-media_da09e4"
      >
        <a
          href="https://www.facebook.com/"
          target="_blank"
        >
          <img
            alt="Facebook"
            src="/src/assets/icons/facebook.svg"
          />
        </a>
        <a
          href="https://www.instagram.com/"
          target="_blank"
        >
          <img
            alt="Instagram"
            src="/src/assets/icons/instagram.svg"
          />
        </a>
        <a
          href="https://www.snapchat.com/"
        >
          <img
            alt="Snapchat"
            src="/src/assets/icons/snapchat.svg"
          />
        </a>
        <a
          href="https://line.me/en/"
        >
          <img
            alt="Line"
            src="/src/assets/icons/line.svg"
          />
        </a>
      </div>
      <div
        class="_payment-method_da09e4"
      >
        <img
          alt="Visa"
          src="/src/assets/icons/visa.svg"
        />
        <img
          alt="American Express"
          src="/src/assets/icons/americanexpress.svg"
        />
        <img
          alt="Google Wallet"
          src="/src/assets/icons/googlewallet.svg"
        />
        <img
          alt="JCB"
          src="/src/assets/icons/jcb.svg"
        />
      </div>
    </div>
    <footer
      class="_footer_da09e4"
    >
      Â© 2024 DL
    </footer>
  </section>
</div>
`;
